### PR TITLE
nf-core bump-version : also bump version of exported conda environment

### PR DESCRIPTION
**Pull request by phue**
Many thanks to contributing to nf-core/tools!

Please fill in the appropriate checklist below (delete whatever is not relevant). These are the most common things requested on pull requests (PRs).

## PR checklist
 - [x] This comment contains a description of changes (with reason)
 - [x] If you've fixed a bug or added code that should be tested, add tests!
 - [x] Documentation in `docs` is updated
 - [x] `CHANGELOG.md` is updated
 - [x] `README.md` is updated

**Learn more about contributing:** https://github.com/nf-core/tools/tree/master/.github/CONTRIBUTING.md

At first, I added pattern matches for each of the occurences in the
Dockerfile, however this looked kinda ugly because there were multiple
'Updating version in Dockerfile ..' lines printed.

So I refactored the code a little to make sure there is only one INFO line printed per file that was changed, followed by all changes in that file.

While I was at it, I also added a bit of color to the output.

### CODE REVIEW

**Comment by ewels** [comment]
Nice! Whilst you’re there, we also need a new lint test to check that this line is present and correct 😅

**Comment by phue** [comment]
> Nice! Whilst you’re there, we also need a new lint test to check that this line is present and correct sweat_smile

good point :+1:

**Comment by phue** [comment]
Seems like the Travis breakage is not related to this, but is also my fault:

I ambitiously drafted a release tag over at `nf-core/methylseq` without knowing that github does create tags for drafts as well. This then ends up in the `pipelines.json`, causing the failures because the `published_at` field is empty for that tag

**Comment by ewels** [comment]
Aha! Sounds like something that I should fix on the website. Can you make an issue for me on the website repo please?

**Comment by ewels** [comment]
Though actually the website front-end code seems fine, so I guess I added logic to handle this. So maybe we just need to fix the tools cli code if the contents of `pipelines.json` make sense..

File: README.md [hunk 1]
```diff
@@ -400,33 +400,32 @@ INFO: Changing version number:
   Current version number is '1.0dev'
   New version number will be '1.0'
 
+
```
Comment (MaxUlysse) [inline README.md]: I'm not sure extra empty lines are OK with MD linting.
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -400,33 +400,32 @@ INFO: Changing version number:
...
 INFO: Updating version in nextflow.config
  - version = '1.0dev'
  + version = '1.0'
 
+
```
Comment (MaxUlysse) [inline README.md]:
```suggestion
```

File: README.md [hunk 3]
```diff
@@ -400,33 +400,32 @@ INFO: Changing version number:
...
 INFO: Updating version in nextflow.config
- - container = 'nfcore/mypipeline:dev'
- + container = 'nfcore/mypipeline:1.0'
+ - process.container = 'nfcore/mypipeline:1.0dev'
+ + process.container = 'nfcore/mypipeline:1.0'
+
```
Comment (MaxUlysse) [inline README.md]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -400,33 +400,32 @@ INFO: Changing version number:
...
 
 INFO: Updating version in .travis.yml
- - docker tag nfcore/mypipeline:dev nfcore/mypipeline:latest
- + docker tag nfcore/mypipeline:dev nfcore/mypipeline:1.0
+ - - docker tag nfcore/mypipeline:dev nfcore/mypipeline:1.0dev
+ + - docker tag nfcore/mypipeline:dev nfcore/mypipeline:1.0
 
-INFO: Updating version in Singularity
- - VERSION 1.0dev
- + VERSION 1.0
```
Comment (MaxUlysse) [inline README.md]:
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -400,33 +400,32 @@ INFO: Changing version number:
...
 
 INFO: Updating version in environment.yml
  - name: nf-core-mypipeline-1.0dev
  + name: nf-core-mypipeline-1.0
 
-INFO: Updating version in Dockerfile
- - PATH /opt/conda/envs/nf-core-mypipeline-1.0dev/bin:$PATH
- + PATH /opt/conda/envs/nf-core-mypipeline-1.0/bin:\$PATH
```
Comment (MaxUlysse) [inline README.md]:
```suggestion
```

File: README.md [hunk 6]
```diff
@@ -405,28 +405,22 @@ INFO: Updating version in nextflow.config
  + version = '1.0'
 
 INFO: Updating version in nextflow.config
- - container = 'nfcore/mypipeline:dev'
- + container = 'nfcore/mypipeline:1.0'
+ - process.container = 'nfcore/mypipeline:1.0dev'
```
Comment (ewels) [inline README.md]: Containers should be `dev`, not `1.0dev`

File: README.md [hunk 7]
```diff
@@ -405,28 +405,22 @@ INFO: Updating version in nextflow.config
...
+ + process.container = 'nfcore/mypipeline:1.0'
 
 INFO: Updating version in .travis.yml
- - docker tag nfcore/mypipeline:dev nfcore/mypipeline:latest
- + docker tag nfcore/mypipeline:dev nfcore/mypipeline:1.0
-
-INFO: Updating version in Singularity
- - VERSION 1.0dev
- + VERSION 1.0
+ - - docker tag nfcore/mypipeline:dev nfcore/mypipeline:1.0dev
```
Comment (ewels) [inline README.md]: How come these docker tags are changing?

File: docs/lint_errors.md [hunk 8]
```diff
@@ -201,11 +201,14 @@ to create the container. Such `Dockerfile`s can usually be very short, eg:
 
 ```Dockerfile
 FROM nfcore/base:1.7
+MAINTAINER Rocky Balboa <your@email.com>
```
Comment (ewels) [inline docs/lint_errors.md]: 🤟🏻

File: nf_core/bump_version.py [hunk 9]
```diff
@@ -113,14 +115,20 @@ def update_file_version(filename, lint_obj, pattern, newstr, allow_multiple=Fals
...
+    if len(matches_pattern) > 1 and not allow_multiple:
         raise SyntaxError("Found more than one version number in {}: '{}'".format(filename, pattern))
 
     # Replace the match
-    logging.info("Updating version in {}\n - {}\n + {}".format(filename, matches[0], newstr))
     new_content = re.sub(pattern, newstr, content)
+    matches_newstr = re.findall("^.*{}.*$".format(newstr),new_content,re.MULTILINE)
+
+    logging.info("Updating version in {}\n".format(filename) +
+        click.style(" - {}\n".format("\n - ".join(matches_pattern).strip()), fg='red') +
```
Comment (ewels) [inline nf_core/bump_version.py]: Nice 👍🏻

**Comment by phue** [comment]
> Is the example output with the changed docker tags intentional, or a manual copy and paste error?

Just a copypaste messup in indeed, good spot :+1:

**Pull request merged**